l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Akretion - Renato Lima <renato.lima@akretion.com.br>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import requests
+
+from odoo import models, fields, api, _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+from odoo.exceptions import UserError
+
+
+class ResCurrencyRateProviderBCB(models.Model):
+    _inherit = 'res.currency.rate.provider'
+
+    service = fields.Selection(
+        selection_add=[('BCB', 'Brazilian Central Bank')])
+
+    @api.model
+    def _get_supported_currencies(self):
+        self.ensure_one()
+        if self.service == 'BCB':
+            url = ('https://olinda.bcb.gov.br/olinda/servico/PTAX/versao'
+                   '/v1/odata/Moedas?$top=100&$format=json&$select=simbolo')
+
+            currencies = []
+            response = requests.get(url)
+            if response.ok:
+                data = response.json()
+                currencies = [x.get('simbolo') for x in data.get('value')]
+
+            return currencies
+        return super()._get_supported_currencies()
+
+    @api.model
+    def _obtain_rates(self, base_currency, currencies, date_from, date_to):
+        self.ensure_one()
+        if self.service == 'BCB':
```

Comment:
This may be "optimized" as in https://github.com/OCA/currency/blob/12.0/currency_rate_update/models/res_currency_rate_provider_ECB.py to avoid extra padding